client/syntaxes/coq.tmLanguage

```diff
@@ -45,7 +45,7 @@
 
         <dict>
             <key>match</key>
-            <string>(Lemma|Theorem|Goal|Example)\s+([a-zA-Z][a-zA-Z0-9_']*)</string>
+            <string>(Lemma|Theorem|Goal|Example)\s+([\p{L}-]([\p{L}-]|[0-9]|[_'])+)</string>
```

Comment:
I'm not very familiar with vscode's regexps, what does the first hyphen mean in the pattern?

Reply:
`\p{L}` matches `à`: I've just left out the end of the range, which would be the last unicode character. It's not a vscode-specific regex: you can look up any unicode regex [reference sheet](https://www.regular-expressions.info/refunicode.html).

Reply:
But you want to match only letters, don't you?

Reply:
More concretely, I'm wondering why `[\p{L}-]` rather than `\p{L}`.

Reply:
`θIO` is a valid identifier: getting it to highlight correctly was the main motivation behind this patch.

Reply:
Think of `[\p{L}-]` as the unicode equivalent of `[a-]` (or `[a-z]`). You don't want every identifier to begin with an `à`, right?

Reply:
Doesn't `b` also match `\p{L}`?

Reply:
Your documentation points to https://www.regular-expressions.info/unicode.html#category, which says: `\p{L} or \p{Letter}: any kind of letter from any language`

Reply:
Thanks for catching this: I read the documentation in a hurry; fixed now.